### DISCUSSION TITLE
Add manifest to frontend, webapp can then act as normal app on Android.

### DISCUSSION THREAD

**Discussion by bluhb** [music-assistant/support#3128] [Feature requests and ideas]
Hello, 
I am now using Music Assistant as my main music player at home. Only difficulty I have is the non-existent app or mobile control.
But I think there would be an "easy"  solution for this. On android, at least, it is possible to have a web app manifest added to the main html file. If the web view is then added to the homescreen, the app will run in a full screen and separate chrome browser. This acts as an independent app on the phone. 
This will prevent chrome opening a new tab every time you press on a pinned icon at the homescreen. I think it then would even be possible to capture device controls, but that is a whole other feature request. 

I was trying if I could find how to add this myself to try it out but I can't seem to find it that easy/good(as I am also using Docker and I am not that proficient in making/adjusting docker containers).

I am very curious if this would be possible and the opinion about this from other people! I would also be prepared to help with a solution of course!

Thanks in advance,
Bluhb

**Comment by OzGav** [comment]
You can submit a PR if you like so this can be looked at by the devs

**Reply by bluhb** [reply]
But for a PR I would have to have the changed files/working solution right? Because at this moment I know theoretically how to do it, but I don't know how to change files in the docker container to test things out exactly :/

If someone can point me in the right direction to change the html files in the docker container I am running, I will try to make this solution and make a PR for it?

**Reply by OzGav** [reply]
I’m not sure it’s as simple as that but @marcelveldt will hopefully find some time to comment.

**Comment by bluhb** [comment]
So i looked some more into this. Apparently there is a manifest file in the docker and to have it work as a standalone app, there would be a need for a https connection. This is forced from how the manifest works. 
What I found is to add
`"launch_handler":{"client_mode":"focus-existing"}`
to the manifest file. This makes sure that every time you open the shortcut it will open in the same tab instead of opening a new tab in chrome. This makes it already working much better.
Next step is to try it when hosted in a https capable way.
For now this is closed I think.
